Fixes CompositeContainer's use of exceptions

mecha opened the pull request:
[`CompositeContainer`] now uses [`NotFoundException`] and [`ContainerException`] correctly.

Aims to incorporate a solution for #9 

[`CompositeContainer`]: https://github.com/Dhii/di/blob/fix/composite-container-incorrect-exceptions_%239/src/CompositeContainer.php

[`ContainerException`]: https://github.com/Dhii/di/blob/fix/composite-container-incorrect-exceptions_%239/src/CompositeContainer.php#L6

[`NotFoundException`]: https://github.com/Dhii/di/blob/fix/composite-container-incorrect-exceptions_%239/src/CompositeContainer.php#L7

mecha committed:
Added tests for exception creation methods (#9)

mecha committed:
Fixed use of exception classes (#9)

mecha committed:
Fixed code standards

XedinUnknown commented on test/functional/CompositeContainerTest.php:
```diff
@@ -337,4 +345,34 @@ public function testThreeLevelComplexRetrieval()
+        $expected = new NotFoundException('test message', 3, null);
+        $exception = $subject->this()->_createNotFoundException('test message', 3, null);
+
+        $this->assertEquals($expected, $exception);
```
I feel it would make more sense to `assertInstanceOf()`. Thoughts?

mecha replied:
Perhaps.

In that case a separate test will be needed to confirm that the `message`, `code` and `previous` properties are correctly set. These are currently tested using the above equivalence check - which checks all three _and_ the instance type.

XedinUnknown replied:
Go ahead. You just need to test `getMessage()`, `getCode()`, and `getPrevious()`.

XedinUnknown replied:
Assert [`ExceptionInterface`](https://github.com/Dhii/di-interface/blob/master/src/ExceptionInterface.php) too.

mecha committed:
Improved test for exception creation methods (#9)

The test now checks using `assertInstanceof()` and separately checks the message, code and previous properties.
